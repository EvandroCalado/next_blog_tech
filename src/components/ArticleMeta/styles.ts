import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child)::after {
      content: ', ';
    }

    span:first-child {
      margin-right: 0.5rem;
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
