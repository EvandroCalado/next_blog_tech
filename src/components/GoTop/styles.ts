import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
    color: ${theme.colors.white};
    position: fixed;
    bottom: 3rem;
    right: 5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    z-index: 6;
    text-decoration: none;
    transition: all 300ms ease-in-out;

    &:hover {
      color: ${theme.colors.secondary};
    }
  `}
`;
