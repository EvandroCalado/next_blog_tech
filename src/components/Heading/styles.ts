import styled, { css, DefaultTheme } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
    font-weight: 900;
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
    font-weight: 900;
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
    font-weight: 900;
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhuge};
    font-weight: 900;
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primary : theme.colors.white};
    line-height: 1.2;
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
    text-align: center;
  `}
`;
