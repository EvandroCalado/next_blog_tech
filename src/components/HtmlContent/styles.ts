import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};

    p {
      margin: ${theme.spacings.xxlarge} 0;
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }

    a:hover {
      filter: brightness(50%);
    }

    img {
      max-width: 100%;
    }

    .image {
      background-color: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.xxlarge} 0;
    }

    .image figcaption {
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3rem;
    }

    .image-style-side {
      float: right;
      margin: ${theme.spacings.medium};
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }

    ul,
    ol {
      margin: ${theme.spacings.xxlarge};
    }

    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }

    table {
      width: 100%;
      border-collapse: collapse;
    }

    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }

    @media ${theme.media.medium} {
      font-size: 2rem;
    }
  `}
`;
