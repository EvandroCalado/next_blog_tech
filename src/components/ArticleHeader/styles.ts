import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.small} {
        font-size: ${theme.font.sizes.medium};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};

    @media ${theme.media.small} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
