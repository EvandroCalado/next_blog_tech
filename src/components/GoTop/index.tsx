import * as Styled from './styles';
import { KeyboardArrowUp } from '@styled-icons/material-outlined/KeyboardArrowUp';

const GoTop = () => {
  const handleClick = () => {
    window.scrollTo(0, 0);
  };

  return (
    <Styled.Container
      onClick={handleClick}
      aria-label="Go to top"
      title="Go to top"
    >
      <KeyboardArrowUp />
    </Styled.Container>
  );
};

export default GoTop;
