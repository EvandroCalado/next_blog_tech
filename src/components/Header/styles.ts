import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.dark};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;

    ${Title} {
      margin: ${theme.spacings.small} 0;
      text-align: left;
    }

    @media ${theme.media.medium} {
      flex-flow: column wrap;
    }
  `}
`;

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;

    @media ${theme.media.medium} {
      margin: 0;
    }
  `}
`;
