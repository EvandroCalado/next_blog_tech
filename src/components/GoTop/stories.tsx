import GoTop from '.';
import { Meta, Story } from '@storybook/react/types-6-0';

export default {
  title: 'GoTop',
  component: GoTop,
  args: {
    children: 'GoTop',
  },
  argTypes: {
    children: { type: 'string' },
  },
} as Meta;

export const Template: Story = (args) => {
  return (
    <div style={{ height: '400vh' }}>
      <h1>Lorem ipsum dolor sit amet.</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi eius optio
        debitis quam maiores odit, soluta amet ducimus. Ipsam quia est voluptas
        quibusdam delectus temporibus omnis consequuntur perspiciatis laborum
        aliquam?
      </p>
      <h1>Lorem ipsum dolor sit amet.</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi eius optio
        debitis quam maiores odit, soluta amet ducimus. Ipsam quia est voluptas
        quibusdam delectus temporibus omnis consequuntur perspiciatis laborum
        aliquam?
      </p>
      <h1>Lorem ipsum dolor sit amet.</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi eius optio
        debitis quam maiores odit, soluta amet ducimus. Ipsam quia est voluptas
        quibusdam delectus temporibus omnis consequuntur perspiciatis laborum
        aliquam?
      </p>
      <h1>Lorem ipsum dolor sit amet.</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi eius optio
        debitis quam maiores odit, soluta amet ducimus. Ipsam quia est voluptas
        quibusdam delectus temporibus omnis consequuntur perspiciatis laborum
        aliquam?
      </p>
      <h1>Lorem ipsum dolor sit amet.</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi eius optio
        debitis quam maiores odit, soluta amet ducimus. Ipsam quia est voluptas
        quibusdam delectus temporibus omnis consequuntur perspiciatis laborum
        aliquam?
      </p>
      <h1>Lorem ipsum dolor sit amet.</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi eius optio
        debitis quam maiores odit, soluta amet ducimus. Ipsam quia est voluptas
        quibusdam delectus temporibus omnis consequuntur perspiciatis laborum
        aliquam?
      </p>
      <h1>Lorem ipsum dolor sit amet.</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi eius optio
        debitis quam maiores odit, soluta amet ducimus. Ipsam quia est voluptas
        quibusdam delectus temporibus omnis consequuntur perspiciatis laborum
        aliquam?
      </p>
      <h1>Lorem ipsum dolor sit amet.</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi eius optio
        debitis quam maiores odit, soluta amet ducimus. Ipsam quia est voluptas
        quibusdam delectus temporibus omnis consequuntur perspiciatis laborum
        aliquam?
      </p>
      <h1>Lorem ipsum dolor sit amet.</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi eius optio
        debitis quam maiores odit, soluta amet ducimus. Ipsam quia est voluptas
        quibusdam delectus temporibus omnis consequuntur perspiciatis laborum
        aliquam?
      </p>
      <h1>Lorem ipsum dolor sit amet.</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi eius optio
        debitis quam maiores odit, soluta amet ducimus. Ipsam quia est voluptas
        quibusdam delectus temporibus omnis consequuntur perspiciatis laborum
        aliquam?
      </p>
      <h1>Lorem ipsum dolor sit amet.</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi eius optio
        debitis quam maiores odit, soluta amet ducimus. Ipsam quia est voluptas
        quibusdam delectus temporibus omnis consequuntur perspiciatis laborum
        aliquam?
      </p>
      <h1>Lorem ipsum dolor sit amet.</h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi eius optio
        debitis quam maiores odit, soluta amet ducimus. Ipsam quia est voluptas
        quibusdam delectus temporibus omnis consequuntur perspiciatis laborum
        aliquam?
      </p>
      <GoTop {...args} />
    </div>
  );
};
