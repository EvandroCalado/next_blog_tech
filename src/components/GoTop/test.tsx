import GoTop from '.';
import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

beforeEach(() => {
  Object.defineProperty(window, 'scrollTo', {
    value: jest.fn(),
    writable: true,
  });
});

describe('<GoTop />', () => {
  it('should scroll to top when clicked', () => {
    const { container } = renderTheme(<GoTop />);
    const button = screen.getByLabelText('Go to top');
    const scrollToSpy = jest.spyOn(window, 'scrollTo');
    fireEvent.click(button);
    expect(scrollToSpy).toHaveBeenCalledWith(0, 0);
    expect(container).toMatchSnapshot();
  });
});
