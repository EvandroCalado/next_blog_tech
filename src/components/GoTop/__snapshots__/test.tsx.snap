// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoTop /> should render a go to top button 1`] = `
.c0 {
  background-color: #f2f2f2;
  color: #2a3840;
  border: 1px solid #2a3840;
  border-radius: 50%;
  position: fixed;
  bottom: 2rem;
  right: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 5rem;
  height: 5rem;
  z-index: 6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

<div>
  <a
    aria-label="Go to top"
    class="c0"
    href="#"
    title="Go to top"
  >
    <svg
      aria-hidden="true"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
      />
    </svg>
  </a>
</div>
`;
